server/services/nextpayment.service.js

```diff
@@ -1,8 +1,18 @@
+import mongoose from 'mongoose';
 import NextPayment from '../models/nextPayment.model';
 import { ErrorHandler } from '../helpers/errorHandler';
 import httpStatus from '../helpers/httpStatus';
+// eslint-disable-next-line import/no-cycle
+import { getOfferById, generatePaymentSchedules } from './offer.service';
+// eslint-disable-next-line import/no-cycle
+import { getTotalTransactionByOfferId } from './transaction.service';
 
-export const getNextPaymentById = async (id) => NextPayment.findById(id).select();
+const { ObjectId } = mongoose.Types.ObjectId;
+
+export const getLastPendingPayment = async (offerId) =>
```

Comment:
Not sure of the number of payments we would have later so let's try to be more explicit by using NextPayment for all payments involving the next payment service.
```suggestion
export const getLastPendingNextPayment = async (offerId) =>
```